sys/mips/include/atomic.h

```diff
@@ -882,6 +1021,229 @@ atomic_fetchadd_64(__volatile uint64_t *p, uint64_t v)
 }
 #endif
 
+
+#ifdef __CHERI_PURE_CAPABILITY__
+/*
+ * cheri-only variants to perform atomic operations on pointers.
+ */
+
+static __inline void
+atomic_set_cap(__volatile uintptr_t *p, uintptr_t v)
```

Comment:
Use _ptr or a uintcap_t type, but don't mix things like this (I prefer the former, as is done on RISC-V).

Reply:
Done the former, mirroring RISC-V.